### PR TITLE
Eigenlayer integration

### PR DESCRIPTION
**Pull request by samparsky**
Integrate Eigenlayer adapters

### CODE REVIEW

File: src/eigenlayer/ObolEigenLayerPodController.sol [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    /// @dev this is the pod manager transparent proxy
+    IEigenPodManager public immutable eigenLayerPodManager;
+
+    /// @notice address of delay withdrawal router
+    IDelayedWithdrawalRouter public immutable delayedWithdrawalRouter;
+
+    /// @notice fee address
+    address public immutable feeRecipient;
+
+    /// @notice fee share
```
Comment (OisinKyne) [inline src/eigenlayer/ObolEigenLayerPodController.sol:58]:
```suggestion
    /// @notice fee share. Represented as an integer from 1->10000 (100%)
```

File: src/eigenlayer/ObolEigenLayerPodController.sol [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        feeRecipient = recipient;
+        feeShare = share;
+        eigenLayerDelegationManager = delegationManager;
+        eigenLayerPodManager = IEigenPodManager(eigenPodManager);
+        delayedWithdrawalRouter = IDelayedWithdrawalRouter(withdrawalRouter);
+    }
+
+    /// @dev Enables contract to receive ETH
+    // defined on the clone implementation
+    // receive() external payable {}
```
Comment (OisinKyne) [inline src/eigenlayer/ObolEigenLayerPodController.sol:100]: What's the story here with commenting out a payable receive? Important/not important? to keep/remove?

File: src/eigenlayer/ObolEigenLayerPodController.sol [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        delayedWithdrawalRouter.claimDelayedWithdrawals(
+            address(this),
+            numberOfDelayedWithdrawalsToClaim
+        );
+
+        // transfer eth to split
+        uint256 balance = address(this).balance;
+        if (feeShare > 0) {
+            uint256 fee =  (balance * feeShare) / PERCENTAGE_SCALE;
+            feeRecipient.safeTransferETH(fee);
```
Comment (OisinKyne) [inline src/eigenlayer/ObolEigenLayerPodController.sol:153]:
```suggestion
            feeRecipient.safeTransferETH(fee);
```
Assumes this handles units properly (same unit as `address().balance`)

File: src/eigenlayer/ObolEigenLayerPodController.sol [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            split.safeTransferETH(balance -= fee);
+        } else {
+            split.safeTransferETH(address(this).balance);
+        }
+    }
+
+    /// @notice Rescue stuck tokens
```
Comment (OisinKyne) [inline src/eigenlayer/ObolEigenLayerPodController.sol:160]:
```suggestion
    /// @notice Rescue stuck tokens by sending them to the split contract.
```

Should we consider something `onlyOwner` that instead can go to an arbitrary address? Maybe wondering if it matters (e.g. a rebase token? is that the only danger?). This contract isn't directly the withdrawal address for a validator right? the odds it gets airdropped tokens should be relatively low I guess? Probably don't have to worry.

**Review by OisinKyne — APPROVED** [review]
Some small updates to comments and some questions. 

This only works for taking a fee inclusive of principal, does it? I haven't looked into the delayedwithdrawal impl enough to figure out could it be like a lido thing where you only get your net amount not the gross amount through this contract.

File: src/eigenlayer/ObolEigenLayerPodControllerFactory.sol [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            feeRecipient,
+            feeShare,
+            delegationManager,
+            eigenPodManager,
+            withdrawalRouter
+        );
+
+        controllerImplementation.initialize(
+            feeRecipient,
+            feeRecipient
```
[the remark this answers is not in the document]
Comment (samparsky) [reply src/eigenlayer/ObolEigenLayerPodControllerFactory.sol:49]: yes, that's correct

File: src/eigenlayer/ObolEigenLayerPodController.sol (hunk 2 above)
Comment (samparsky) [reply src/eigenlayer/ObolEigenLayerPodController.sol:100]: it's implemented on the clone. this is meant to be documentation for integrations

File: src/eigenlayer/ObolEigenLayerPodController.sol (hunk 3 above)
Comment (samparsky) [reply src/eigenlayer/ObolEigenLayerPodController.sol:153]: yes, that's correct

**Pull request merged**